fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexScrubber.java

```diff
@@ -938,20 +936,13 @@ private void validateIndex() {
             if (!metaData.hasIndex(index.getName()) || index != metaData.getIndex(index.getName())) {
                 throw new MetaDataException("Index " + index.getName() + " not contained within specified metadata");
             }
-            if (recordTypes == null) {
-                recordTypes = metaData.recordTypesForIndex(index);
-            } else {
+            if (recordTypes != null) {
                 for (RecordType recordType : recordTypes) {
                     if (recordType != metaData.getIndexableRecordType(recordType.getName())) {
                         throw new MetaDataException("Record type " + recordType.getName() + " not contained within specified metadata");
                     }
                 }
             }
-            if (recordTypes.stream().anyMatch(RecordType::isSynthetic)) {
-                // The (stored) types to scan, not the (synthetic) types that are indexed.
-                recordTypes = new SyntheticRecordPlanner(metaData, new RecordStoreState(null, null), getTimer())
-                        .storedRecordTypesForIndex(index, recordTypes);
-            }
```

Comment:
This change is somewhat subtle. Here, we validate that if the `recordTypes` are set that they are in the meta-data. We leave the `recordTypes` field as null, but that will get set by `IndexingCommon.fillTargetIndexers` when the `IndexingCommon` gets constructed. I needed to do this because previously, this was resulting in `IndexingContexts` being created that did not have `isSynthetic` set even if the index was on synthetic types